### PR TITLE
Fix invalid kcompact free call and invalid csched rbt sizing

### PR DESCRIPTION
**Pull request by beckerg**
Signed-off-by: Greg Becker <gbecker@micron.com>

### CODE REVIEW

File: lib/cn/csched_sp3.c [hunk 1]
```diff
@@ -189,7 +189,7 @@ struct sp3 {
     atomic_int               running;
     struct sp3_qinfo         qinfo[SP3_QNUM_MAX];
 
-    struct rb_root rbt[wtype_MAX - 1] HSE_L1D_ALIGNED;
+    struct rb_root rbt[wtype_MAX] HSE_L1D_ALIGNED;
```
Comment (beckerg) [inline lib/cn/csched_sp3.c:192]: I'd prefer not to talk about this...

File: lib/cn/csched_sp3.h [hunk 2]
```diff
@@ -32,7 +32,7 @@ struct sp3_rbe {
 };
 
 struct sp3_node {
-    struct sp3_rbe   spn_rbe[wtype_MAX - 1];
+    struct sp3_rbe   spn_rbe[wtype_MAX];
```
Comment (beckerg) [inline lib/cn/csched_sp3.h:35]: Nor this...

**Review by nabeelmmd — APPROVED** [review]

**Review by tristan957 — APPROVED** [review]
We can forgive you this time Greg, but next time you're booted off the island! :)

**Review by gsramdasi — APPROVED** [review]

**Comment by beckerg** [comment]
I'm starting to worry that it's a telltale sign of dementia...

**Pull request merged**